### PR TITLE
logging package integration - DUNE-DAQ/logging#1

### PR DESCRIPTION
**Pull request by ron003**
I have or will have PRs in for the ron003/logging_issue_1_example_integration branch in 6 packages:
appfwk
dfmodules
ipm
networkqueue
readout
trigemu

With these packages on the ron003/logging_issue_1_example_integration branch, along with the other develop
branches for cmdlib, daq-cmake, dataformats, dfmessages, logging restcmd, serialization packages as of
16 feb and with adjustments to the dunedaq-v2.2.0 development environment:

sed -i 's/v4_3_1b e19/v4_3_1c e19:prof/' dbt-settings
for pp in "msgpack_c v3_3_0 e19:prof" "cppzmq v4_3_0 e19:prof";do
  grep "$pp" dbt-settings || sed -ie "/zmq/a \
    \"$pp\"
" dbt-settings
done

sed -i 's/"dataformats"/"serialization" "networkqueue" "dataformats"/' sourcecode/dbt-build-order.cmake

Then, the minidaq demonstration works.

Other changes still need to be made as indicated (not exhaustively) by the output of
`find sourcecode -type f | xargs egrep 'ERS_(INFO|LOG|DEBUG)'

### CODE REVIEW

**Comment by philiprodrigues** [comment]
I've merged the latest develop into this branch, and updated a few more places where `TLVL_TRACE` appeared. I also removed all the `TRACE_NAME` macro definitions, since they all matched the filenames. @eflumerf @jcfreeman2 could one of you take a look and merge if everything looks OK please?

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,14 +6,16 @@ find_package(daq-cmake REQUIRED)
 daq_setup_environment()
 
 find_package(appfwk REQUIRED)
+find_package(logging REQUIRED)
 
 find_package(cppzmq REQUIRED)
 find_package(ers REQUIRED)
```
Comment (eflumerf) [inline CMakeLists.txt:12]: Is this still needed?

**Review by eflumerf — APPROVED** [review]
LGTM

**Comment by ron003** [comment]
Not absolutely *needed* (i.e. required), but I would lean toward keeping based on the way the c++ lint does not like people to
rely on nested includes. However, I talked to John and he leans the other way based on the logging is fundamental and
one could always rely on it being brought in be appfwk.   At this point, it's up to you.
--Ron
On Fri, 2021-03-05 at 09:12 -0800, eflumerf wrote:

@eflumerf commented on this pull request.

________________________________

In CMakeLists.txt<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_DUNE-2DDAQ_ipm_pull_22-23discussion-5Fr588485744&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=7KXkqf0bzyUQv15jBKEcHw&m=HWRbgmPTUkGbq6cTJi5Gtxnl7ngeH-v_uWwRO2iQxl0&s=L7Q-Itit6wu9UKRr7g4qTRV1BEP_WYhOS1WFzAvhjQk&e=>:

> @@ -6,14 +6,16 @@ find_package(daq-cmake REQUIRED)

 daq_setup_environment()



 find_package(appfwk REQUIRED)

+find_package(logging REQUIRED)



 find_package(cppzmq REQUIRED)

 find_package(ers REQUIRED)

Is this still needed?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_DUNE-2DDAQ_ipm_pull_22-23pullrequestreview-2D605432991&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=7KXkqf0bzyUQv15jBKEcHw&m=HWRbgmPTUkGbq6cTJi5Gtxnl7ngeH-v_uWwRO2iQxl0&s=NggaR6ky3GmwfzPch5TF8belDW45kxUXHUZrae26fNc&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AGH7VZQTMJUKF53UKE4OXN3TCEGJHANCNFSM4XXCEH3A&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=7KXkqf0bzyUQv15jBKEcHw&m=HWRbgmPTUkGbq6cTJi5Gtxnl7ngeH-v_uWwRO2iQxl0&s=9_2vPPVGyeKLO-Y53d8PLFQwUp-aPkjPhChJZJJK4v8&e=>.

--

Ron Rechenmacher
Engineer, Tools and Advanced Computing
Fermi National Accelerator Laboratory
Batavia, IL  60510

**Comment by eflumerf** [comment]
I was actually trying to tag the `find_package(ers REQUIRED)`

**Comment by jcfreeman2** [comment]
One other thought that occurred to me is that we can draw crude parallels with our traditional artdaq suite. E.g., if an experiment's package is based on artdaq, in its product_deps file it doesn't need to specify artdaq-core since that gets brought in by artdaq. Also, the "fwk" in appfwk suggests that it brings in a bunch of useful tools for the DUNE DAQ software developer, which can include logging functionality.

**Pull request merged**